### PR TITLE
Basic Backend API

### PR DESCRIPTION
**Pull request by falk-werner**
This pull requests introduces a basic backend API.
_Basic_ means that not all backend functions of [note.py](https://github.com/falk-werner/note.py) are available here, but enough to create a basic frontend. The following functions are available:

- list all notes
- create a note
- change title of a note
- read the contents of a note
- write the contents of a note
- delete a note

The following functions are left out:

-  take a screenshot
- handling tags

A configuration file named `.notejs.yml` is used to specify the directory of the notes. If not present $home/notes will be used. The layout of the notes directory is compatible to the one we used in `note.py`.

### CODE REVIEW

**Comment by falk-werner** [comment]
Also note that this pull request contains some kind of frontend. This frontend is intended for testing purposes. All backend functions are available in the frontend. We should replace the frontend once this pull request is finished.

File: src/backend/config.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const os = require("os");
+const path = require("path");
+const fs = require("fs");
+const yaml = require("yaml");
+
+class Config {
+
+    constructor() {
+        const default_config = {
+            base_path: "{home}/notes"
```
Comment (hmlampe) [inline src/backend/config.js:10]: here "/notes" is added the first time

File: src/backend/config.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        const template = this.config.base_path;
+        const homedir = os.homedir();
+        const base_path = template.replaceAll("{home}", homedir);
+
+        return base_path;
+
+    }
+
+    get note_path() {
+        return path.join(this.#base_path, "notes");
```
Comment (hmlampe) [inline src/backend/config.js:43]:
```suggestion
        return this.#base_path;
```
here "/notes" is added a second time, which results in "home/notes/notes

File: src/backend/config.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        if (!fs.existsSync(this.note_path)) {
+            fs.mkdirSync(this.note_path, {recursive: true});
+        }
+    }
+
+    #load_config() {
+        try {
+            const homedir = os.homedir();
+            const filename = path.join(homedir, ".notejs.yml");
```
[the remark this answers is not in the document]
Comment (falk-werner) [reply src/backend/config.js:23]: Exactly. This pull request introduces the backend API. It is not meant to provide a complete implementation.
I suggest to separate writing the config file from the backend API. In note.py, we write the config file when note.py is closed. Therefore, there is no API needed to trigger saving the config file.

File: src/backend/config.js (hunk 1 above)
Comment (falk-werner) [reply src/backend/config.js]: The duplicate "notes" directory comes from `note.py`. There we have a base directory `.note.py` containing a subdirectory `notes`. I've simply renames `.note.py` to `notes`. I'm in favor of your suggestion.

**Review by hmlampe — APPROVED** [review]

**Pull request merged**